mso/resource_mso_schema_site_service_graph.go

```diff
@@ -61,10 +61,116 @@ func resourceMSOSchemaSiteServiceGraph() *schema.Resource {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"none",
+								"redir",
+							}, false),
+							Default: "none",
+						},
+						"provider_connector_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "none",
+						},
+						"consumer_interface": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"provider_interface": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
 					},
 				},
 			},
 		}),
+
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+			/* This function validates the user input for service_node.provider_connector_type when
+			the template_service_graph.service_node.type is 'other' or 'firewall'.
+
+			- The user input for site_service_graph.service_node.other_provider_connector_type should be one of 'none' or 'redir',
+			when the corresponding template_service_graph.service_node.type is 'other'.
+
+			- The user input for site_service_graph.servicenode.firewall_provider_connector_type_list should be one of 'none', 'redir', 'snat', 'dnat' or 'snat_dnat',
+			when the corresponding template_service_graph.service_node.type is 'firewall'.
+			*/
+
+			// Create a list of service node types using the user input(template service graph).
+			msoClient := v.(*client.Client)
+			_, schemaId := diff.GetChange("schema_id")
+			_, templateName := diff.GetChange("template_name")
+			_, graphName := diff.GetChange("service_graph_name")
+			cont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", schemaId))
+			if err != nil {
+				return err
+			}
+
+			sgCont, _, err := getTemplateServiceGraphCont(cont, templateName.(string), graphName.(string))
+			if err != nil {
+				log.Printf("graphcont err %v", err)
+				return err
+			}
+
+			var templateServiceNodeList []string
+			serviceNodes := sgCont.S("serviceNodes").Data().([]interface{})
+			for _, val := range serviceNodes {
+				serviceNodeValues := val.(map[string]interface{})
+				nodeId := models.StripQuotes(serviceNodeValues["serviceNodeTypeId"].(string))
+
+				nodeType, err := getNodeNameFromId(msoClient, nodeId)
+				if err != nil {
+					return err
+				}
+
+				templateServiceNodeList = append(templateServiceNodeList, nodeType)
+			}
+
+			/* Loop trough the templateServiceNodeList and validate the site level user input(provider_connector_type)
+			   to verify it's value for nodetype 'other' and 'firewall'. */
+			_, siteServiceNodes := diff.GetChange("service_node")
+
+			for _, templateNodeType := range templateServiceNodeList {
+				found := false
+				for i, val := range siteServiceNodes.([]interface{}) {
+					serviceNode := val.(map[string]interface{})
+					if templateNodeType == "other" {
+						var other_provider_connector_type_list = []string{"none", "redir"}
```

Comment:
Is there a reason to initiate the variable different compared to firewall_provider_connector_type_list?

```suggestion
						other_provider_connector_type_list := []string{"none", "redir"}
```